lib/linguist/heuristics.yml

```diff
@@ -709,6 +709,10 @@ disambiguations:
   - language: Markdown
     # Markdown syntax for scdoc
     pattern: '^#+\s+(NAME|SYNOPSIS|DESCRIPTION)'
+- extensions: ['.script']
+  rules:
+  - language: URScript
+    pattern: '^\s*def\s+[a-zA-Z_][a-zA-Z0-9_]*\s*\([^)]*\)\s*:[\s\S]*\s+end+\s'
```

Comment:
This regex is vulnerable to ReDoS attack.

Reply:
I see. I will try to find something that is not suffering from that and is still able to capture the URScript. 

Again thank you for your feedback🙏

Reply:
I updated the regex and tested it with https://devina.io/redos-checker 

Regarding the specificity of the regex. I try to match against all other samples in the project, and got the following results:  
```
@EbbeFuglsang ➜ /workspaces $ ruby search.rb ./linguist/samples '^\s*def\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*\(([^)]*)\)\s*:(((?:(?!^\send($|\s))[\s\S])*)\send($|\s))'
Could not read file ./linguist/samples/Altium Designer/Sample Schematic Sheet.SchDoc: invalid byte sequence in UTF-8
Could not read file ./linguist/samples/Oberon/beNull.ob2: invalid byte sequence in UTF-8
Could not read file ./linguist/samples/Roff/roff.1in: invalid byte sequence in UTF-8
Match found in: ./linguist/samples/URScript/admittance_control.script
Match found in: ./linguist/samples/URScript/math.script
Match found in: ./linguist/samples/URScript/payload_get_struct.script
Match found in: ./linguist/samples/URScript/payload_manipulators.script
Match found in: ./linguist/samples/URScript/payload_set.script
Match found in: ./linguist/samples/URScript/payload_tests.script
Match found in: ./linguist/samples/jq/builtin.jq
Match count: 7
Files count: 3047
```
If the samples is representative for all other code, the amount of false positive is not too bad.